Mathlib/Order/Hom/WithTopBot.lean

```diff
@@ -63,6 +63,34 @@ def coeOrderHom {α : Type*} [Preorder α] : α ↪o WithTop α where
   inj' := WithTop.coe_injective
   map_rel_iff' := WithTop.coe_le_coe
 
+/-- Removing then adding ⊤ makes the type `OrderIso` to the original type. -/
+def subtypeOrderIso [DecidableEq α] [PartialOrder α] [OrderTop α] :
```

Comment:
slightly more general:
```suggestion
def subtypeOrderIso [PartialOrder α] [OrderTop α] [DecidablePred (· = (⊤ : α))] :
```
and should we have the WithBot version too?

Reply:
The WithBot version is below